openrarity/collection/collection.py

```diff
@@ -0,0 +1,337 @@
+import json
+from hashlib import md5
+from itertools import chain
+from logging import Logger
+from pathlib import Path
+from typing import Literal, TypedDict, cast, overload
+
+from satchel.aggregate import groupapply
+
+from openrarity.io import read, write
+from openrarity.metrics.ic import calculate_entropy, information_content
+from openrarity.token import (
+    AttributeStatistic,
+    RankedToken,
+    RawToken,
+    TokenId,
+    TokenStatistic,
+    ValidatedTokenAttribute,
+    enforce_schema,
+    validate_tokens,
+)
+from openrarity.token.metadata.dtypes import (
+    process_numeric_dtypes,
+    process_string_dtypes,
+)
+from openrarity.types import JsonEncodable
+from openrarity.utils import merge, rank_over
+
+from .utils import aggregate_tokens, flatten_token_data
+
+logger = Logger(__name__)
+
+
+class AttributesStatistics(TypedDict):
+    string: list[AttributeStatistic]
+    number: list[AttributeStatistic]
+    date: list[AttributeStatistic]
+
+
+class TokenCollection:
+    """ """
```

Comment:
?

Reply:
Just a placeholder. Needs the docstring completed